New component to handle dynamic form creation

pserquiz opened the pull request:
# Critical Changes

# Changes
A component that will handle dynamic form creation based on the given fieldset

# Issues Closed

pserquiz committed:
New component to handle form creation

This component should handle dynamic form creation based on the given fieldset

pserquiz committed:
Design attr label changes

cwico commented on force-app/main/default/lwc/appFieldSet/appFieldSet.js-meta.xml:
```diff
@@ -10,8 +10,8 @@
-			<property name="fieldSetApiName" type="String" label="What is the fieldSet api" description="The api name of the fieldset that will be used"/>
+			<property name="fieldSetObject" type="String" label="The Object Api Name" description="Which object has the fieldset"/>
+			<property name="fieldSetApiName" type="String" label="The FieldSet Api Name" description="The api name of the fieldset that will be used"/>
 			<property name="isUsedInCommunityBuilder" type="Boolean" label="Check if its being used in the community builder" description="This is component is embed to a parent component?" default="false"/>
```
Why do you need this? Is this suppose to be the flag for standalone form? If so can you call it isStandalone=true as default.

pserquiz replied:
thats to be used as design attributes, this component can be dragged and drop in the community builder too.

cwico commented on force-app/main/default/classes/AppFieldSetController.cls:
```diff
@@ -0,0 +1,159 @@
+	 * @param sObjectApiName the object api name
+	 */
+	@AuraEnabled
+	public static void insertSObject(String jSONSObject, String sObjectApiName) {
```
some formatting but this looks great.

cwico commented on force-app/main/default/lwc/appFieldSet/appFieldSet.js:
```diff
@@ -0,0 +1,157 @@
+	}
+
+	//Support methods
+	buildObjectToInsert() {
```
loooks great! how can we support namespaced field?

cwico replied:
I think we should make this more generic like getRecord()?

pserquiz replied:
it builds based on the fields we get from the backend, it should handle namespace already

Commit:
Merge 3c15d447468f8fecfa30cf1dc5aafcb0c1cdfefe into feature/field-set-component

pserquiz committed:
Renamed variables and deleted one design attribute

cwico reviewed (approved):
awesome

pserquiz committed:
Changes for handling EDIT

This changes will make sure that the component also handles EDIT a record for the given recordId

Commit:
Merge 0bbd62a3779af20e4cd13b07f9eb9ae69e4e00c0 into feature/field-set-component

Commit:
Merge 9118c2d36021757c27b41b18d1850bbe712e2234 into feature/field-set-component

Commit:
Merge 97fffa54919b9bdd7242349acfe72ac24b1ac938 into feature/field-set-component

camgood-trac commented:
This is not ready to merge yet.

@pserquiz @cwico to finish up

Commit:
Merge e6ef96733e33d58e240ee5fb7b7749bf669537f7 into feature/field-set-component

Commit:
Merge 92f141fdd5c408b491c3ce342dc34328b2197a8f into feature/field-set-component

pserquiz committed:
Added lookup type fields handler to the component

pserquiz committed:
Added support to list of field names

The component now also handles the load of a given list of field names

Commit:
Merge 9fc85cfc9dc5c04441029b8da50d181ee521e160 into feature/field-set-component

cwico commented on force-app/main/default/classes/AppFieldSetController.cls:
```diff
@@ -0,0 +1,315 @@
+		// Execute search query
+		String query = 'FIND \'' + String.escapeSingleQuotes(searchTerm) + '\' IN ALL FIELDS RETURNING ' + lookupObject + '(Id, Name WHERE Id NOT IN :selectedIds) LIMIT :MAX_RESULTS';
+		system.debug('query => '+query);
+		List<List<SObject>> searchResults = Search.query(query);
```
I don't think we need double lists here anymore. Can you take that out?

cwico commented on force-app/main/default/classes/AppFieldSetController.cls:
```diff
@@ -0,0 +1,315 @@
+							lookupObject,
+							icon,
+							name,
+							lookupObject+' • ' + name
```
Remove or comment the **' • ' + name** for now or replace name with id

cwico commented on force-app/main/default/classes/AppFieldSetController.cls:
```diff
@@ -0,0 +1,315 @@
+		SObject newSobject = targetType.newSObject();
+		Map<String, Schema.sObjectField> targetFields = targetType.getDescribe().fields.getMap();
+		for (String key : fieldMap.keySet())
+		{
```
braces formatting

cwico commented on force-app/main/default/classes/AppFieldSetControllerTest.cls:
```diff
@@ -0,0 +1,112 @@
+ * @author  Pedro Serquiz, Traction on Demand
+ * @date    2020-04-19
+ */
+@IsTest
```
try Parallel?

cwico commented on force-app/main/default/lwc/appFieldSet/appFieldSet.html:
```diff
@@ -0,0 +1,58 @@
+		<template for:each={fieldSetData} for:item="field">
+			<div key={field.fieldPath} class="slds-col slds-small-size_1-of-1 slds-medium-size_1-of-2 slds-large-size_1-of-2">
+				<template if:true={field.isLookupField}>
+					<c-lookup
```
yay

cwico commented on force-app/main/default/lwc/appFieldSet/appFieldSet.js:
```diff
@@ -0,0 +1,363 @@
+	renderedCallback() {
+		if (!this.isFirstRender) {
+			this.isFirstRender = true;
+			getOrgNamespace().then(result => {
```
Smart, should we create a base component that we can start using for namespace and other stuff. Namespace for now.

pserquiz replied:
Yeah, good idea, but I dont think we need this for now, maybe have a ticket so we dont forget later on.

cwico commented on force-app/main/default/lwc/appUtils/appUtils.js:
```diff
@@ -24,9 +24,16 @@ const getFieldValue = (obj, fieldName, namespace) => {
 
-export {handleToastMessage, getOrgNamespace, getFieldValue, applyNamespace}
\ No newline at end of file
+export {handleToastMessage, getOrgNamespace, getFieldValue, setNamespaceToList}
```
Why replace applyNamespace with setNamespaceToList? when applyNamespace works with lists also and any nested objects and fields. Can you double check this?

pserquiz replied:
sorry, my branch was behind, did a pull and will fix that now, nice catch

pserquiz replied:
fixed o/

cwico commented on force-app/main/default/lwc/appFieldSet/appFieldSet.js:
```diff
@@ -0,0 +1,363 @@
+		const selection = this.template.querySelector('c-lookup').getSelection();
+		if (selection.length === 0) {
+			this.lookupConfig.lookupErrors = [
+				{ message: 'Please choose your care facility to continue...' }
```
Can you change this label?  to please choose your "OBJECT with name (if possible)"   to continue

cwico reviewed (changes requested):
There are some notes that I put on here. Needs some minor rivisions.

pserquiz committed:
Merge branch 'master' into feature/field-set-component

pserquiz committed:
Changes for Cwico review

cwico reviewed (approved):
Looks good
